### DISCUSSION TITLE
Allow asset bundle class to insert js/css code directly into template instead registering them as file links

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#14046] [Ideas] [answered]
In some cases it is more advantageous to just insert js/css into template. 
One example would be to avoid making additional server request to external file.

**Comment by brandonkelly** [comment, marked as answer]
You can already do that using `Craft::$app->view->registerJs()` and `registerCss()`.

**Reply by piotrpog** [reply]
This methods take in css and js code directly. I propose adding some method that would allow for working with asset bundle class. So just like we use `Craft::$app->view->registerAssetBundle()`, we would also be able to use something like `Craft::$app->view->registerAssetBundleInline()`.

**Reply by piotrpog** [reply]
Of course we can also loop through js or css assets of this class, the grab contents of this files and finally pass these contents to `registerJs` or `registerCss`, but it would be more comfortable to have ready to use method.

**Reply by brandonkelly** [reply]
The whole point of the AssetManager and asset bundles is for publishing static asset files from `vendor/` into the web root, so it wouldn’t make sense to shoehorn JS script registration into that. Plus it would go against best practices: any JS code that should be included statically (without being generated dynamically at all) is best off being included via a JS file, which the browser can cache and optimize.
